### PR TITLE
[RISCV] Assign different scheduling classes for VMADC/VMSBC

### PR DESCRIPTION
**Pull request by mshockwave**
Split the scheduling classes of VMADC/VMSBC away from that of VADC/VSBC. Because the former are technically mask-producing instructions rather than normal vector arithmetics, which might have different performance characteristics on some processors.

This is effectively NFC.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVInstrInfoV.td [hunk 1]
```diff
@@ -630,31 +630,43 @@ multiclass VMRG_IV_V_X_I<string opcodestr, bits<6> funct6> {
...
                         forceMasked=0>;
   def X : VALUVXNoVm<funct6, OPIVX, opcodestr # ".vx">,
-          SchedBinaryMC<"WriteVICALUX", "ReadVICALUV", "ReadVICALUX",
+          SchedBinaryMC<WritePrefix#"X", "ReadVICALUV", "ReadVICALUX",
                         forceMasked=0>;
 }
 
 multiclass VALUNoVm_IV_V_X_I<string opcodestr, bits<6> funct6>
    : VALUNoVm_IV_V_X<opcodestr, funct6> {
+  // if LSB of funct6 is 1, it's a mask-producing instruction that
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVInstrInfoV.td:665]: It only has `vm*` users, so I think we don't need to this change. You can simply change `WriteVICALUI` to `WriteVICALUMI`.

File: llvm/lib/Target/RISCV/RISCVInstrInfoV.td [hunk 2]
```diff
@@ -630,31 +630,43 @@ multiclass VMRG_IV_V_X_I<string opcodestr, bits<6> funct6> {
...
+  // uses a different scheduling class.
+  defvar WriteSched = !if(funct6{0}, "WriteVICALUMI", "WriteVICALUI");
   def IM : VALUmVI<funct6, opcodestr # ".vim">,
-           SchedUnaryMC<"WriteVICALUI", "ReadVICALUV">;
+           SchedUnaryMC<WriteSched, "ReadVICALUV">;
 }
 
 multiclass VALUNoVm_IV_V_X<string opcodestr, bits<6> funct6> {
+  // if LSB of funct6 is 1, it's a mask-producing instruction that
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVInstrInfoV.td:652]: Same here.

File: llvm/lib/Target/RISCV/RISCVInstrInfoV.td (hunk 1 above)
Comment (mshockwave) [reply llvm/lib/Target/RISCV/RISCVInstrInfoV.td]: good catch! It's fixed now.

File: llvm/lib/Target/RISCV/RISCVInstrInfoV.td (hunk 2 above)
Comment (mshockwave) [reply llvm/lib/Target/RISCV/RISCVInstrInfoV.td]: Fixed.

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Pull request merged**